provider/pkg/provider/previewOutputs_test.go

```diff
@@ -0,0 +1,416 @@
+package provider
+
+import (
+	"testing"
+
+	"github.com/pulumi/pulumi-aws-native/provider/pkg/metadata"
+	"github.com/pulumi/pulumi/pkg/v3/codegen/schema"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestPrevierOutputs(t *testing.T) {
+	t.Run("Nested output value", func(t *testing.T) {
+		result := PreviewOutputs(
+			resource.NewPropertyMapFromMap(map[string]interface{}{
+				"name": "my-access-point",
+				"objectLambdaConfiguration": map[string]interface{}{
+					"allowedFeatures":          []string{"GetObject-Range"},
+					"cloudWatchMetricsEnabled": true,
+				},
+			}),
+			map[string]metadata.CloudAPIType{
+				"aws-native:s3objectlambda:AccessPointAlias": {
+					Properties: map[string]schema.PropertySpec{
+						"Status": {
```

Comment:
This one is actually "status" in Pulumi schema, is there a reason we are capitalizing it here?